lib/services/publish-service.ts

```diff
@@ -0,0 +1,211 @@
+
+                // (id - 1) is the index of the object to be deleted
+                _.remove(this.allPublishConnections, this.ftpPublishConnections[id - 1]);
+                message = util.format("Connection '%s' deleted successfully", this.ftpPublishConnections[id - 1].name);
+            } else {
+                var name = idOrName;
+                var connectionsToBeRemoved = _.filter(this.allPublishConnections, function(connection) {
+                        return connection.name === name;
+                });
+
+                if (connectionsToBeRemoved.length > 1) {
+                    this.$logger.info("Multiple connections with name '%s' found." + os.EOL + 
+                                        "Type in the index of the one you want to delete " + os.EOL + 
+                                        "Or 0 if you want to delete them all", name);
+
+                    this.listConnections(connectionsToBeRemoved);
+                    var ftpConnectionId = parseInt(this.$prompter.getString("Index:").wait());
+                    if (isNaN(ftpConnectionId) || ftpConnectionId > connectionsToBeRemoved.length || ftpConnectionId < 0) {
+                        this.$errors.failWithoutHelp("Invalid index");
+                    }
+
+                    if (ftpConnectionId === 0) {
+                        _.remove(this.allPublishConnections, function(connection) {
+                                return connection.name === name;
+                            });
+                        message = util.format("Connections named '%s' deleted successfully", name);
+                    } else {
+                        // again (id - 1) is the index of the object to be deleted
+                        _.remove(this.allPublishConnections, connectionsToBeRemoved[ftpConnectionId - 1]);
+                        message = util.format("Connection '%s' deleted successfully", connectionsToBeRemoved[ftpConnectionId - 1].name);
+                    }
+                } else if (connectionsToBeRemoved.length === 1) {
+                    _.remove(this.allPublishConnections, connectionsToBeRemoved[0]);
+                    message = util.format("Connection '%s' deleted successfully", connectionsToBeRemoved[0].name);
+                } else {
+                    this.$errors.failWithoutHelp("No connections named '%s' found", name);
+                }
+            }
+
+            this.savePublishJsonFile();
+            this.$logger.info(message);
+        }).future<void>()();
+    }
+
+    private readConnections(): IFuture<void> {
+        return (() => {
+            var projectDir = this.$project.getProjectDir().wait();
+            if (projectDir) {
+                var publishFilePath = path.join(projectDir, PublishService.JSON_PUBLISH_FILE_NAME);
+                try {
+                    this.allPublishConnections = this.$fs.readJson(publishFilePath).wait();
+                } catch (err) {
+                    if (err.code === 'ENOENT') {
```

Comment:
With `this.$fs.tryExecuteFileOperation` I won't have access to `err.toString()` which provides invaluable information in case the json project file is messed up